generator.js

```diff
@@ -93,23 +65,46 @@ async function getSVGFile(icon, version) {
     .replace('>', '>{props.title && <title>{props.title}</title>}')
     .replace(/style="enableBackground:(.*?);?"/g, 'enableBackground="$1"')
     .replace(/style="fill:(.*?);?"/g, 'fill="$1"')
+    .replace(/xmlns:xlink/g, 'xmlnsXlink')
+    .replace(/xlink:href/g, 'xlinkHref')
+}
+
+async function generateComponent(icons) {
+  for (let i = 0; i < icons.length; i++) {
+    const icon = icons[i]
+
+    for (let j = 0; j < ICON_FAMILIES.length; j++) {
+      const family = ICON_FAMILIES[j]
+
+      const name = formatName(icon.name, family.namePostfix)
+      const svg = await getSVGFile(icon.name, family.id, icon.version)
+
+      console.log(`Downloading ${name} (${i + 1}/${icons.length})`)
+
+      await fs.writeFileSync(
+        path.join(__dirname, 'src', `${name}.tsx`),
+        componentTemplate(name, svg)
+      )
+    }
+  }
 }
 
-/**
- * Main function
- */
 ;(async () => {
-  // Setup source folder
-  setup()
+  generateIconPropsFile()
+
+  const res = await axios.get(GOOGLE_FONTS_URL)
 
-  // Get icon information from google fonts
-  const res = await axios.get(iconURL)
   // remove )]}' from the response
   const data = res.data.substring(4)
-
-  // Parse response from json to js object
   const icons = await JSON.parse(data)
 
-  // Generate icon components and Index
-  await generateComponent(icons.icons)
+  const iconChunks = new Array(
+    Math.ceil(icons.icons.length / DOWNLOAD_CHUNK_SIZE)
+  )
+    .fill()
+    .map(() => icons.icons.splice(0, DOWNLOAD_CHUNK_SIZE))
```

Comment:
Why do you do the split up into multiple chunks? I don't really see any benefit here how you handle it.

Reply:
I have cleaned it up